doc/ir-semantics.text

```diff
@@ -2,34 +2,42 @@
 
 x        variables
 
-P ::= Segment  active   ↦ S, (Ds ↦ S)*   program:            active version and others
+P ::= Function main    ↦ F, (Df ↦ F)*    program:            main function and other functions
+F ::= Segment  active  ↦ S, (Ds ↦ S)*    function:           active version and others
 S ::= start ↦ i,          I              segment:            instruction stream with dedicated start
 I ::= (l ↦ i)*                           instruction stream: labeled instructions
 
 Ds ::= Segment  s                  segment declaration
 
+f  function name
 s  optimization version
 l  labels
 
+main      main function
 active    active segment
 start     start label
 
 a  addresses
 
 i ::=    instructions
-| const x = c                   constant variable
-| mut x = c                     mutable variable
-| x <- c                        assignment
+| const x = ce                  constant variable
+| mut x = ce                    mutable variable
+| x <- ce                       assignment
 | branch e l₁ l₂                conditional
 | goto l                        goto
 | print e                       print
+| return e                      return
```

Comment:
Oh and why is the call restricted to the assignments, there is no good rationale.

I certainly do not want the call to be a normal expression for now, because then expressions are not pure anymore (ie. no `call() + call()`).

In the case of `print e` it could be ce as well. I thought it might be easier to reason if you cannot do a side-effect and a call in one instruction.